shared/js/background/popupMessaging.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @typedef {import('webextension-polyfill').Runtime.Port} Port
+ * @typedef {import('@duckduckgo/privacy-dashboard/schema/__generated__/schema.types').IncomingExtensionMessage} OutgoingPopupMessage
+ */
+
+// Messaging connection with the popup UI (when active).
+/** @type {Port?} */
+let activePort = null
+
+/**
+ * Set up the messaging connection with the popup UI.
+ *
+ * Note: If the ServiceWorker dies while there is an open connection, the popup
+ *       will take care of re-opening the connection.
+ *
+ * @param {Port} port
+ * @param {Record<
+ *     string,
+ *     (options: any, sender: any, message: any) => any
+ * >} messageHandlers
+ */
+export function popupConnectionOpened (port, messageHandlers) {
```

Comment:
_(Message handlers passed in here instead of imported, to reduce circular dependencies.)_